pygmt/tests/test_clib_put_strings.py

```diff
@@ -8,12 +8,31 @@
 from pygmt import clib
 from pygmt.exceptions import GMTCLibError
 from pygmt.helpers import GMTTempFile
+from pygmt.helpers.testing import skip_if_no
```

Comment:
Also need to revert changes in this PR, since the low-level function `Session.put_strings` only accepts `numpy.str_` and everything else should be converted to `numpy.str_` by the `_to_numpy` function before calling `put_strings`.

Reply:
Ok, just to document things, the way we are intending pyarrow.string inputs to go through high-level modules like `fig.text` is:

1. `pygmt.Figure.text` -> `clib.Session.virtualfile_in` -> `clib.Session.virtualfile_from_vectors`
2. In `clib.Session.virtualfile_from_vectors`:
   - First call `clib.conversion.vectors_to_arrays` -> `clib.conversion._to_numpy` -> `np.ascontiguous_array` to convert `pyarrow.string` to `np.str_`
   - With a `list[np.str_]`, this is then passed to `clib.Session.put_strings`

For the record, the original intention in this PR, mostly in commit d379e4667c5f8fcc907a3a56543ae734b8c605eb, was to let `clib.Session.put_strings` handle both `np.str_` and `pyarrow.string`.